homeassistant/components/switch/rest.py

```diff
@@ -97,10 +112,22 @@ def turn_off(self, **kwargs):
 
     def update(self):
         """Get the latest data from REST API and update the state."""
-        request = requests.get(self._resource, timeout=10)
-        if request.text == self._body_on:
-            self._state = True
-        elif request.text == self._body_off:
-            self._state = False
+        request = requests.get(self._resource, timeout=self._timeout)
+
+        if self._value_template is not None:
+            response = template.render_with_possible_json_value(
+                self._hass, self._value_template, request.text, 'None')
+            response = response.lower()
+            if response == 'true':
```

Comment:
Wouldn't it be confusing to the user that when a template value is given, the `body on` and `body off` values are getting ignored.

By introducing template rendering for outgoing body on/body off we should probably have a separate conf entry for data that comes back from the polling for state.